core/src/impl/Kokkos_TaskNode.hpp

```diff
@@ -70,6 +70,14 @@
 namespace Kokkos {
 namespace Impl {
 
+#ifdef KOKKOS_COMPILER_PGI
+// Bizzarely, an extra jump instruction forces the PGI compiler to not have a
+// bug related to (probably?) empty base optimization and/or aggregate
+// construction.  This must be defined out-of-line to generate a jump
+// jump instruction
+void _kokkos_pgi_compiler_bug_workaround();
```

Comment:
Any reason not to have the definition here so we don't introduce the workaround in `Kokkos_Core.cpp` as well?

Reply:
Yeah, second sentence of the comment. If we put it inline, the optimizer will eliminate it before it reaches the buggy part of the PGI optimizer, at least presumably. I haven’t checked though, so I could be wrong.